src/TelemetryChannels/ServerTelemetryChannel/Shared/AdaptiveSamplingTelemetryProcessor.cs

```diff
@@ -0,0 +1,262 @@
+            }
+
+            set
+            {
+                this.estimatorSettings.MaxSamplingPercentage = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets duration of the sampling percentage evaluation 
+        /// interval in seconds.
+        /// </summary>
+        public int EvaluationIntervalSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.EvaluationIntervalSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.EvaluationIntervalSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to decrease
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageDecreaseTimeoutSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.SamplingPercentageDecreaseTimeoutSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.SamplingPercentageDecreaseTimeoutSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to increase
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageIncreaseTimeoutSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.SamplingPercentageIncreaseTimeoutSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.SamplingPercentageIncreaseTimeoutSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets exponential moving average ratio (factor) applied
+        /// during calculation of rate of telemetry items produced by the application.
+        /// </summary>
+        public double MovingAverageRatio
+        {
+            get
+            {
+                return this.estimatorSettings.MovingAverageRatio;
+            }
+
+            set
+            {
+                this.estimatorSettings.MovingAverageRatio = value;
+            }
+        }
+
+        /// <summary>
+        /// Processes telemetry item.
+        /// </summary>
+        /// <param name="item">Telemetry item to process.</param>
+        public void Process(ITelemetry item)
+        {
+            this.samplingProcessor.Process(item);
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        /// <param name="disposing">True if disposing.</param>
+        protected virtual void Dispose(bool disposing)
```

Comment:
Use ```disposing``` in code. Alternatively seal the class and do not create this protected method at all.